### PR TITLE
Fix EventSourceGenerator generated file extension

### PR DESCRIPTION
**Pull request by stephentoub**
Contributes to https://github.com/dotnet/runtime/issues/51058

### CODE REVIEW

**Review by danmoseley — APPROVED** [review]
This and the Interop generator should also probably be emitting `[GeneratedCodeAttribute]` like the other generators.

File: src/libraries/System.Private.CoreLib/generators/EventSourceGenerator.Emitter.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@ public void Emit(EventSourceClass[] eventSources, CancellationToken cancellation
                     _builder.AppendLine("using System;");
                     GenType(ec);
 
-                    _context.AddSource($"{ec.ClassName}.Generated", SourceText.From(_builder.ToString(), Encoding.UTF8));
+                    _context.AddSource($"{ec.ClassName}.g.cs", SourceText.From(_builder.ToString(), Encoding.UTF8));
```
Comment (ViktorHofer) [inline src/libraries/System.Private.CoreLib/generators/EventSourceGenerator.Emitter.cs:35]: Why wasn't it "Generated.cs" before if the file extension is required?

Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/generators/EventSourceGenerator.Emitter.cs:35]: it's not required, at least not from a source gen perspective. it's convention.

Comment (danmoseley) [reply src/libraries/System.Private.CoreLib/generators/EventSourceGenerator.Emitter.cs:35]: the file on disk is only for debugging purposes, so I assumed it simply didn't have .cs before. but it seems I'm wrong:
 `...generated\System.Private.CoreLib.Generators\Generators.EventSourceGenerator\RuntimeEventSource.Generated.cs`

Comment (danmoseley) [reply src/libraries/System.Private.CoreLib/generators/EventSourceGenerator.Emitter.cs:35]: I'm guessing with this change it's `.g.cs.cs` ?

Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/generators/EventSourceGenerator.Emitter.cs:35]: > I'm guessing with this change it's .g.cs.cs ?

no, roslyn appends ".cs" if you don't include it yourself

Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/generators/EventSourceGenerator.Emitter.cs:35]: https://sourceroslyn.io/#Microsoft.CodeAnalysis/SourceGeneration/AdditionalSourcesCollection.cs,129

**Comment by danmoseley** [comment]
Should I open issues for `[GeneratedCodeAttribute]` or do you believe it's immaterial? Not sure what it triggers.

**Comment by stephentoub** [comment]
> Should I open issues for [GeneratedCodeAttribute]

Sure, it's good hygiene.

**Review by ViktorHofer — APPROVED** [review]

**Comment by danmoseley** [comment]
Opened https://github.com/dotnet/runtime/issues/64541

**Pull request merged**